pennylane_lightning_kokkos/lightning_kokkos.py

```diff
@@ -210,12 +212,25 @@ def var(self, observable, shot_range=None, bin_size=None):
     def expval(self, observable, shot_range=None, bin_size=None):
         if observable.name in [
             "Projector",
-            "Hamiltonian",
-            "SparseHamiltonian",
         ]:
             self.syncD2H()
             return super().expval(observable, shot_range=shot_range, bin_size=bin_size)
 
+        if observable.name in ["SparseHamiltonian"]:
+            CSR_SparseHamiltonian = observable.sparse_matrix().tocsr()
+            return self._kokkos_state.ExpectationValue(
+                CSR_SparseHamiltonian.data,
+                CSR_SparseHamiltonian.indices,
+                CSR_SparseHamiltonian.indptr,
+            )
+
+        if observable.name in ["Hamiltonian"]:
+            device_wires = self.map_wires(observable.wires)
+            # Since we currently offload hermitian observables to default.qubit, we can assume the matrix exists
```

Comment:
Good spot, this comment is from LightningGPU, it needs to be removed.

Reply:
The comment is still here. :laughing:

Reply:
Yep, I also removed this from LightningGPU in a recent commit. No need for it anywhere anymore.

Reply:
Also, adding a threshold here to use the dense offload if the number of qubits is <13 is a good idea.

We can also, for now, aim to convert the dense to a sparse for larger sizes, to ensure we don't hit the memory wall. This can be alleviated later by adding support for Pauli-word expvals at the CPP layer.

See [here](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/devices/default_qubit.py#L574) for how default.qubit currently does this.

Reply:
Finally got rid of it :)